distribution/sc-proxy-dex/src/proxy_pair.rs

```diff
@@ -374,22 +374,18 @@ pub trait ProxyPairModule: proxy_common::ProxyCommonModule {
             let difference = assets_received - locked_assets_invested.clone();
             self.send()
                 .direct(&caller, &asset_token_id, &difference, &[]);
-            self.send()
-                .esdt_local_burn(&asset_token_id, &locked_assets_invested);
         } else if assets_received < locked_assets_invested {
             let difference = locked_assets_invested - assets_received.clone();
             self.send().esdt_nft_burn(
                 &locked_asset_token_id,
                 attributes.locked_assets_nonce,
                 &difference,
             );
-            self.send()
-                .esdt_local_burn(&asset_token_id, &assets_received);
-        } else {
-            self.send()
-                .esdt_local_burn(&asset_token_id, &assets_received);
         }
 
+        self.send()
```

Comment:
move burn before send tokens. have all these things everywhere before the actual send

Reply:
send and burn operations are the last operations in the contracts. their order does not quite matter since no calls like get_esdt_balance are done within the contracts